chain-impl-mockchain/src/txbuilder.rs

```diff
@@ -289,3 +289,136 @@ impl TransactionFinalizer {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::fee::LinearFee;
+    use crate::transaction::Input;
+    use crate::transaction::NoExtra;
+    use crate::transaction::INPUT_PTR_SIZE;
+    use chain_addr::Address;
+    use quickcheck::Arbitrary;
+    use quickcheck::Gen;
+    use std::ops::RangeBounds;
+
+    quickcheck! {
+        fn tx_builder_creates_balanced_tx(arbitrary_builder: ArbitraryTransactionBuilder) -> () {
+            let builder = arbitrary_builder.0;
+            let policy = OutputPolicy::Forget;
+            let fee = fee_for_tx_builder(&builder);
+
+            let (balance, _transaction) =  builder.finalize(fee, policy).unwrap();
+
+            match balance {
+                Balance::Positive(_) => panic!("Positive balance"),
```

Comment:
don't use panic, quickcheck as a type `TestResult` in which you can provide information